nodes/src/nodes/tool_shell/shell_executor.py

```
@@ -0,0 +1,149 @@
+    max_output_bytes: int,
+) -> dict:
+    """Run *command* in the host shell and capture its result.
+
+    Stdout and stderr are streamed through reader threads and capped at
+    ``max_output_bytes`` per stream so a runaway command cannot exhaust
+    engine memory. Once a stream's buffer reaches the cap, further chunks
+    are drained and discarded so the child can finish writing without
+    blocking on a full OS pipe (preserving its natural exit code).
+    """
+    try:
+        proc = subprocess.Popen(
```

Ruff (0.15.12)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)